tools/azsdk-cli/docs/specs/6-package-updater.spec.md

```diff
@@ -0,0 +1,689 @@
+
+Example aggregated output (successful):
+
+```json
+{
+      "result": "succeeded",
+      "steps": [
+            {"tool":"update-ci", "result":"succeeded", "ci-file": "../ci.yml"},
+            {"tool":"update-changelog", "result":"succeeded"},
+            {"tool":"update-version", "result":"succeeded", "version":"1.2.0", "release-date": "2025-10-17"},
+            {"tool":"update-metadata", "result":"succeeded"}
+      ],
+      "message": "Package updates complete", "next_steps": "open PR"
+}
+```
+
+### User Experience
+
+Typical CLI workflow:
+
+```bash
+azsdk package update-ci --package-path /abs/sdk/path
+azsdk package update-changelog-content --package-path /abs/sdk/path
+azsdk package update-metadata --package-path /abs/sdk/path
+azsdk package update-version --package-path /abs/sdk/path --version 1.0.0 --release-date 2025-10-17
+```
+
+### Architecture Diagram
+
+```mermaid
+flowchart TD
+  Agent[/Agent / User/] --> azsdk_cli
+
+  subgraph azsdk_cli
+    direction TB
+    classifier[SDK Classifier]
+    resolver[Config Resolver]
+    executor[Script Executor]
+    serializer[Result Serializer]
+  end
+
+  classifier --> resolver
+  resolver --> executor
+  executor --> serializer
+
+  style azsdk_cli fill:#e3f2fd,stroke:#0277bd
+  style Agent fill:#e1f5fe,stroke:#0288d1
+```
+
+---
+
+### Workflow Diagram
+
+#### Update-Changelog (mgmt-plane)
+
+```mermaid
+flowchart TD
+      A[Start: UpdateChangelogAsync] --> B[Validate packagePath] 
+      B --> C{packagePath valid?}
+      C -->|No| D[Return failure: Package path required/invalid]
+      C -->|Yes| E[Discover SDK repository root] 
+    
+      E --> F{Repository root found?}
+      F -->|No| G[Return failure: Failed to discover SDK repo]
+      F -->|Yes| H[Get update-changelog-script configuration] 
+    
+      H --> I{Configuration retrieval successful?}
+      I -->|No| J[Return failure: Failed to get update-changelog-script configuration] 
+      I -->|Yes| K{Configuration type?}
+    
+      K -->|Command| L[Prepare update-changelog command] 
+      K -->|ScriptPath| M[Prepare update-changelog script] 
+    
+      L --> N[Execute update-changelog command/script] 
+      M --> N
+    
+      N --> O{update-changelog process completed?} 
+      O -->|Failed| P[Return failure with exit code and output]
+      O -->|Success| Q[Return success]
+    
+      style A fill:#e1f5fe
+      style Q fill:#c8e6c9
+      style D fill:#ffcdd2
+      style G fill:#ffcdd2
+      style J fill:#ffcdd2
+      style P fill:#ffcdd2
+```
+
+> **Note:** Other tools (version update, metadata update, CI update) follow a similar flowchart pattern.
+
+---
+
+## Open Questions
+
+---
+
+## Success Criteria
+
+This tool set is complete when:
+
+- [ ] All four tools emit structured JSON with consistent schema.
+- [ ] Management-plane package run updates version + changelog automatically.
```

Comment:
but here we would need to list the limitations or exceptions too.
might be better to either call out specifically every language per category or summarize that all languages and categories that are not in the exceptions should work.

Reply:
Added exceptions.